src/time-picker/TimePicker.js

```diff
@@ -0,0 +1,211 @@
+  const hours = parseInt(inputArray[0]);
+  const minutes = parseInt(inputArray[1]);
+  if (hours !== 0 && numberRegex.test(hours) && numberRegex.test(minutes)) {
+    const optionTime12 = [
+      { value: `${cleanedInput}am`, label: timeFormatter12(hours, minutes) },
+      { value: `${cleanedInput}pm`, label: timeFormatter12(hours + 12, minutes) }
+    ];
+    const optionTime24 = [{ value: cleanedInput, label: timeFormatter24(hours, minutes) }];
+
+    if (timeFormat === H12 && hours <= 12 ) {
+      return optionTime12;
+    } else if (timeFormat === H24) {
+      return optionTime24;
+    }
+  }
+  return [];
+};
+
+const Integer = t.refinement(t.Number, n => n % 1 === 0, 'Integer');
+const Hour = t.refinement(Integer, int => int >= 0 && int <= 23, 'Hour');
+const Minute = t.refinement(Integer, int => int >= 0 && int <= 59, 'Minute');
+const TimeFormat = t.enums.of([H12, H24], 'TimeFormat');
+const Time = t.struct({
+  hours: Hour,
+  minutes: Minute
+}, 'Time');
+
+const Props = t.refinement(t.struct({
+  onChange: t.Function,
+  value: t.maybe(Time),
+  minTime: t.maybe(Time),
+  maxTime: t.maybe(Time),
+  interval: t.maybe(Integer),
+  placeholder: t.maybe(t.String),
+  timeFormat: t.maybe(TimeFormat),
+  id: t.maybe(t.String),
+  className: t.maybe(t.String),
+  style: t.maybe(t.Object)
+}), ({ value, minTime, maxTime }) => (
+  lteTime(minTime, maxTime) && (!value || (lteTime(value, maxTime) && lteTime(minTime, value)))
+), 'Props');
+
+@skinnable()
+@props(Props)
+export default class TimePicker extends React.Component {
+
+  static defaultProps = {
+    placeholder: `--${separator}--`,
+    timeFormat: H24,
+    interval: 30,
+    minTime: { hours: 0, minutes: 0 },
+    maxTime: { hours: 23, minutes: 59 }
+  }
+
+  /**
+    We use filterOptions because seems that is not possible
+    to change the options from outside of this function,
+    then we use this tricky hack and we workaround this creating a new set
+    of options as we wish and filter it if needed, or just returning a brand new array
+  **/
+  createOptionsList = (_, inputStr) => {
+    /**
+    _ is a placeholder because the TLC pass me the optionsList, but i don't really care. YOLO
+    **/
+    const { minTime, maxTime, interval, timeFormat } = this.props;
+    const timeFormatter = timeFormat === H24 ? timeFormatter24 : timeFormatter12;
+    const minHours = minTime.hours;
+    const maxHours = maxTime.hours;
+    const minMinutes = minTime.minutes;
+    const maxMinutes = maxTime.minutes;
+
+    const hours = range(minHours, maxHours + 1);
+    const minutes = range(0, 60, interval);
+    const options = flatten(hours.map(hour => minutes.filter(minute => (
+      (hour !== minHours || minute >= minMinutes) &&
+      (hour !== maxHours || minute <= maxMinutes)
+    )).map( minute => formatValue(timeFormatter, hour, minute)
+    )));
+    return this.filterOrRecreateOptions(options, inputStr);
+  };
+
+  filterOrRecreateOptions = (options, inputStr) => {
```

Comment:
This is the function described above, can be obviously splitted in to function, on for the filter on the other one for the custom generated optionsList.